torchci/pages/api/drci/drci.ts

```diff
@@ -11,17 +11,24 @@ interface PRandJobs {
     jobs: RecentWorkflowsData[];
 }
 
+export default async function handler() {
+    fetchWorkflows();
+}
+
 export async function fetchWorkflows() {
     const recentWorkflows: RecentWorkflowsData[] = await fetchRecentWorkflows(
         NUM_MINUTES + ""
     );
 
     const workflowsByPR = reorganizeWorkflows(recentWorkflows);
 
-    for (const [pr_number, pr] of workflowsByPR) {
-        const { pending, failedJobs } = getWorkflowAnalysis(pr);
-        const failureInfo = constructFailureAnalysis(pending, failedJobs, pr.head_sha);
-        //add failureInfo to Dr. CI comment on each PR
+    for (const [pr_number, pr_info] of workflowsByPR) {
+        const { pending, failedJobs } = getWorkflowAnalysis(pr_info);
+
+        const failureInfo = constructFailureAnalysis(pending, failedJobs, pr_info.head_sha);
+        const comment = formDrciHeader(pr_number) + failureInfo + DRCI_COMMENT_END;
+
+        updateCommentWithWorkflow(pr_info, comment);
```

Comment:
I think this needs to be awaited 

Also prefer not to use an async in a for loop. It's better to gather all the Promises in an array and use Promise.all(...) to do it. Otherwise you're gonna wait sequentially for all the requests to finish vs waiting for them in parallel.